### PR TITLE
Update documentations for Testing #357

### LINKED ISSUE #357: Update documentations for Testing

### PR DESCRIPTION
**Pull request by m133225**
Fixes #357

### CODE REVIEW

File: docs/Testing.md [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+Tests' settings are mostly contained in `build.gradle` and `.travis.yml`.
+
 ## Gradle Tasks
 There are a few key gradle tasks defined that we can play around with:  
-- `headless` to run **headless testing**
-- `headful` to run **headful testing**
+- `alltests` to run all tests:
+  - `guiTest`
+  - `guiUnitTest`
+  - `unitTest`
```
Comment (damithc) [inline docs/Testing.md]: All these three should be plural?

File: docs/Testing.md [hunk 2]
```diff
@@ -43,7 +48,7 @@ There are a few key gradle tasks defined that we can play around with:
 
 ## CI Testing
```
Comment (damithc) [inline docs/Testing.md]: The command at line 46 does not work anymore

File: docs/Testing.md [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
 ## Local Testing
 ### How to do some common testing-related tasks
 - To run all tests in headless mode: `./gradlew`
-  - It will run `clean`, `headless` tasks.
-- To run checkstyle followed by headful tests `./gradlew checkStyle headful`
-- To run **headless testing** only: `./gradlew headless`
-  - Running this or `headful` repeatedly will not re-run the tests unless the build files are `clean`ed first.
-    e.g. `./gradlew clean headless`
-- To run **headful testing** only: `./gradlew headful`
+  - It will run `clean`, `headless`, `alltests`, `jacocoRootReport` tasks.
```
Comment (damithc) [inline docs/Testing.md]: Shall we use a shorter alias for `jacocoRootReport`?

**Comment by damithc** [comment]
No more comments.

File: docs/Testing.md [hunk 4]
```diff
@@ -1,28 +1,62 @@
-# Running Tests
+# Testing
+
+## Background
+We are using [Gradle](https://docs.gradle.org/)'s [wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) to setup testing configurations and running tests.
+Gradle tasks are run using `gradlew <tasks>` on Windows and `./gradlew <tasks>` on Mac/Linux.
```
Comment (damithc) [inline docs/Testing.md]: No `w` in Mac/Linux command right?

Comment (m133225) [inline docs/Testing.md]: There is.

File: docs/Testing.md [hunk 5]
```diff
@@ -1,28 +1,62 @@
...
+Tests can be found in the `./src/test/java` folder. We have grouped and packaged the tests into the following types.
+
+1. Unit Tests - `address` package
+  - Logic Testing
+
+2. GUI Unit Tests - `guiunittests` package
+    - Tests the UI interaction within a single component, and ensure its behaviour holds.
+
+3. GUI Tests (Integration) - `guitests` package
+  - Tests the UI interaction with the user as well as the interaction between various components (e.g. passing of data)
```
Comment (damithc) [inline docs/Testing.md]: These are actually `system tests`? (i.e. we test the entire system).

Comment (m133225) [inline docs/Testing.md]: In terms of the package naming or the documentation's grouping? (or both 😛 )

**Comment by damithc** [comment]
No other comments.

File: docs/Testing.md (hunk 5 above)
Comment (damithc) [inline docs/Testing.md]: In terms of documentation. 
Anyway, we should take another pass at categorizing tests, probably not right now though.
See http://googletesting.blogspot.sg/2010/12/test-sizes.html

**Pull request merged**